src/main/java/org/traccar/api/resource/NotificationResource.java

```diff
@@ -113,15 +114,27 @@ public Response testMessage(@PathParam("notificator") String notificator)
     public Response sendMessage(
             @PathParam("notificator") String notificator, @QueryParam("userId") List<Long> userIds,
             NotificationMessage message) throws MessageException, StorageException {
-        permissionsService.checkAdmin(getUserId());
+        permissionsService.checkManager(getUserId());
         List<User> users;
         if (userIds.isEmpty()) {
-            users = storage.getObjects(User.class, new Request(new Columns.All()));
+            users = permissionsService.notAdmin(getUserId()) ?
+                    storage.getObjects(User.class, new Request(new Columns.All(),
+                            new Condition.Permission(User.class, getUserId(), ManagedUser.class).excludeGroups())) :
+                    storage.getObjects(User.class, new Request(new Columns.All()));
+
         } else {
             users = new ArrayList<>();
             for (long userId : userIds) {
-                users.add(storage.getObject(
-                        User.class, new Request(new Columns.All(), new Condition.Equals("id", userId))));
+                if (permissionsService.notAdmin(getUserId())) {
+                    var conditions = new LinkedList<Condition>();
+                    conditions.add(new Condition.Equals("id", userId));
+                    conditions.add(new Condition.Permission(User.class, getUserId(), ManagedUser.class).excludeGroups());
+                    users.add(storage.getObject(
+                            User.class, new Request(new Columns.All(), Condition.merge(conditions))));
+                } else {
+                    users.add(storage.getObject(
+                            User.class, new Request(new Columns.All(), new Condition.Equals("id", userId))));
+                }
```

Comment:
It probably makes sense to do something like this:

```suggestion
                    var conditions = new LinkedList<Condition>();
                    conditions.add(new Condition.Equals("id", userId));
                if (permissionsService.notAdmin(getUserId())) {
                    conditions.add(new Condition.Permission(User.class, getUserId(), ManagedUser.class).excludeGroups());
                }
                    users.add(storage.getObject(
                            User.class, new Request(new Columns.All(), Condition.merge(conditions))));
```